src/dotnet/ToolManifest/IToolManifestFinder.cs

```diff
@@ -10,5 +11,6 @@ internal interface IToolManifestFinder
     {
         IReadOnlyCollection<ToolManifestPackage> Find(FilePath? filePath = null);
         FilePath FindFirst();
+        IReadOnlyList<FilePath> FindContainPackageId(PackageId packageId);
```

Comment:
Nit:

```suggestion
        IReadOnlyList<FilePath> FindByPackageId(PackageId packageId);
```

Reply:
Additionally, could this be `IEnumerable<FilePath>` instead of `IReadOnlyList`?  The former would allow us to use an iterator for an implementation.  Just curious.

Reply:
The order matters, (a test assert that as well), so i want to use "List" as a way to convey that.

Reply:
To me, at least, `IReadOnlyList` conveys no notion of stable ordering above what `IEnumerable` offers; you'd need documentation for that.  I would only use `IReadOnlyList` if you need indexing/count.  But I can live with this.

Reply:
Added comment